src/blob/core.rs

```diff
@@ -350,51 +358,61 @@ where
         key: &K,
         meta: Option<&Meta>,
         check_filters: bool,
-    ) -> Result<Option<Entry>> {
+    ) -> Result<ReadResult<Entry>> {
```

Comment:
`read_all_entries` above should also return `ReadResult<>`:
1. At least one non-deleted record -> `ReadResult::Found()`
2. There are records for the key, but the last one is deletion record -> `ReadResult::Deleted()`
3. No records -> `ReadResult::NotFound`